File: 2553.go
import "slices"

func separateDigits(nums []int) []int {
	result := make([]int, 0, len(nums))
	var single []int
	for _, n := range nums {
		for n > 0 {
			single = append(single, n%10)
			n /= 10
		}

		for _, n := range slices.Backward(single) {
			result = append(result, n)
		}

		single = single[:0]
	}

	return result
}

File: 2553.rs
impl Solution {
    pub fn separate_digits(nums: Vec<i32>) -> Vec<i32> {
        let mut result = Vec::with_capacity(nums.len());
        let mut single = Vec::new();

        for mut n in nums {
            while n > 0 {
                single.push(n % 10);
                n = n / 10;
            }

            for n in single.iter().rev() {
                result.push(*n);
            }

            single.clear();
        }

        result
    }
}
